### PR TITLE
Batched draw_lines

### LINKED ISSUE #2351: Add `draw_lines` with batches
Right now the `draw_line` doesn't support batch and only is for 3ch inputs and specify line width too.
-> https://kornia.readthedocs.io/en/latest/utils.html#kornia.utils.draw_line

Would be great to support seamless to `draw_convex_polygon`

### PR DESCRIPTION
**Pull request by cjpurackal**
#### Changes

Fixes #2351 #2188 

#### Type of change

- [ ] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Comment by johnnv1** [comment]
Do these changes fix/cover #2188?

File: kornia/utils/draw.py [hunk 1]
```diff
@@ -47,87 +47,99 @@ def draw_line(image: torch.Tensor, p1: torch.Tensor, p2: torch.Tensor, color: to
...
+            for x in (x_i + s * torch.arange(0, dx_i + 1, dx_i / dy_)[: int(dy_.item()) + 1].ceil().to(image.device))
+        ]
+        rest_y_cords = [
+            y
+            for y_i, s, dy_i in zip(y1[rest_mask], dy_sign[rest_mask], dy[rest_mask])
+            for y in (y_i + s * torch.arange(0, dy_i + 1, 1).to(image.device))
+        ]
+    x_cords = torch.tensor(dx_zero_x_cords + dy_zero_x_cords + dx_gt_dy_x_cords + rest_x_cords).long()
+    y_cords = torch.tensor(dx_zero_y_cords + dy_zero_y_cords + dx_gt_dy_y_cords + rest_y_cords).long()
+    image[:, y_cords, x_cords] = color.view(-1, 1)
```
Comment (edgarriba) [inline kornia/utils/draw.py:142]: Just to think in next steps, how this method will support adjusting the line with; or giving subpixel coordinates?

**Comment by cjpurackal** [comment]
> Do these changes fix/cover #2188?
Seems to work
image

File: kornia/utils/draw.py (hunk 1 above)
Comment (cjpurackal) [reply kornia/utils/draw.py:142]: I missed the width and sub-pixel parts. Will work on that

Comment (johnnv1) [reply kornia/utils/draw.py:142]: Should we merge this as it is, and work on this in a subsequent PR?

File: kornia/utils/draw.py [hunk 2]
```diff
@@ -47,87 +47,99 @@ def draw_line(image: torch.Tensor, p1: torch.Tensor, p2: torch.Tensor, color: to
...
+    rest_x_cords, rest_y_cords = [], []
+
+    if dx_zero_mask.any():
+        dx_zero_x_cords = [
+            x for x_i, dy_i in zip(x1[dx_zero_mask], dy[dx_zero_mask]) for x in x_i.repeat(int(dy_i.item() + 1))
+        ]
+        dx_zero_y_cords = [
+            y
+            for y_i, s, dy_ in zip(y1[dx_zero_mask], dy_sign[dx_zero_mask], dy[dx_zero_mask])
+            for y in (y_i + s * torch.arange(0, dy_ + 1, 1).to(image.device))
```
Comment (edgarriba) [inline kornia/utils/draw.py:105]:
```suggestion
            for y in (y_i + s * torch.arange(0, dy_ + 1, 1, device=image.device))
```

Comment (edgarriba) [reply kornia/utils/draw.py:105]: and same below

**Review by edgarriba — APPROVED** [review]

**Review by johnnv1 — APPROVED** [review]

**Pull request merged**